client/executor/src/wasm_runtime.rs

```diff
@@ -354,41 +418,51 @@ fn create_versioned_wasm_runtime(
 	max_instances: usize,
 	cache_path: Option<&Path>,
 ) -> Result<VersionedRuntime, WasmError> {
-	#[cfg(not(target_os = "unknown"))]
-	let time = std::time::Instant::now();
+	// The incoming code may be actually compressed. We decompress it here and then work with
+	// the uncompressed code from now on.
+	use sp_maybe_compressed_blob::CODE_BLOB_BOMB_LIMIT;
+	let code = sp_maybe_compressed_blob::decompress(code, CODE_BLOB_BOMB_LIMIT)
+		.map_err(|e| WasmError::Other(format!("Decompression error: {:?}", e)))?;
+	let blob = sc_executor_common::runtime_blob::RuntimeBlob::new(&code)?;
+
+	let mut version = None;
+
+	// Use the runtime blob to scan if there is any metadata embedded into the wasm binary pertaining
+	// to runtime version. We do it before consuming the runtime blob for creating the runtime.
+	if let Some(embedded_version) = read_embedded_version(&blob)? {
+		version = Some(embedded_version);
+	}
```

Comment:
```suggestion
	// Use the runtime blob to scan if there is any metadata embedded into the wasm binary pertaining
	// to runtime version. We do it before consuming the runtime blob for creating the runtime.
	let mut version = read_embedded_version(&blob)?;
```

Reply:
Do you insist? I do realize that it is possible and is more succinct. However, IMO, the verison is there emphasizes that embedded version can be None

Reply:
I will not insist, but I find it weird :D 

In the end the type system is emphasizing that this can be `None` :P

Reply:
Ok found a compromise:

```rust
let mut version: Option<_> = read_embedded_version(&blob)?;
```